docs/HOWWEWORK.md

```diff
@@ -0,0 +1,335 @@
+# Kotlin Template
+
+
+1. All new work resigns in `feature/*` branches. If some new interfaces were introduced, remember to cover them with unit-tests.
+2. After your work on the feature branch is complete, open the PR and ask someone who is working on this project (or worked previously) to review it.
+3. When PR is opened, Bitrise will automatically run all the code quality checks and unit-tests and will generate a report.
+4. When PR is reviewed, merged and all the checks have passed, remember to make a new builds.
+
+> Pro-tip: When opening pull-requests and/or creating branches remember to reference the JIRA issue/ticket
+
+## Enterprise level projects
+
+### Code quaulity
+Enterprise level projects also contain `lintrules` module with custom linting rules to maintain code quaulity. Moreover, each module uses `detect` and `klint` gradle plugins to maintain code style and cleanliness. Every pull request made, will launch these checks to make sure develop/master branches are always "clean" and release-ready. You can also run these checks manually using `gradle` commands like so:
+```
+./gradlew detekt
+./gradlew lint
+./gradlew ktlintCheck
+```
+
+### Unit-test coverage
+Enterprise level project are required to have **80% code-coverage** with the most important piece being application's **business logic** (100% coverage) and Presentation/UI layers. Android uses `JaCoCo` gradle plugin to generate coverage reports for all the modules and ties them all together. If there are some files you wish to exclude from coverage-report (Dagger files for example) you can update `jacoco.gradle` to with additional files like so:
+```groovy
+def toExclude = ['**/R.class',
+                 '**/R$*.class',
+                 '**/*$$ViewBinder*.*',
+                 '**/injection/*',
+                 '**/**Builder.*'
+```
+When it comes to writing unit-test itself please refer to our extensive guide for unit testing: https://github.com/nodes-android/guidelines/blob/master/unittesting.md
+
+## Android Project
+
+Android project is based on our newest template for client projects that uses [Google's ViewModels](https://developer.android.com/topic/libraries/architecture/viewmodel) with a lightweight ViewState approach, similar to MVI and  follows multi-modular approach to support clean architecture principles.
+
+
+
+### Build Configuration
+Due to the modular architecture approach top-level `build.gradle` should provides all dependencies versions, android API level,  and API keys that could be shared across different modules using extensions.
+```groovy
+
+ext.sdks = [
+          compileSdkVersion: 28,
+          buildToolsVersion: '28.0.3',
+          minSdkVersion    : 21,
+          targetSdkVersion : 28
+    ]
+
+ext.versions = [
+         kotlin             : '1.3.41',
+         coroutines         : '1.3.0-M2',
```

Comment:
Can be just latestVersion so it gets more generic

Reply:
fixed